src/OpenTelemetry.Instrumentation.ServiceFabricRemoting/ActorRemoting/TraceContextEnrichedActorRemotingClientFactory.cs

```diff
@@ -0,0 +1,105 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0using System.Fabric;
+
+using System.Fabric;
+using Microsoft.ServiceFabric.Actors.Remoting.V2.FabricTransport.Client;
+using Microsoft.ServiceFabric.Services.Client;
+using Microsoft.ServiceFabric.Services.Communication.Client;
+using Microsoft.ServiceFabric.Services.Remoting.FabricTransport;
+using Microsoft.ServiceFabric.Services.Remoting.V2;
+using Microsoft.ServiceFabric.Services.Remoting.V2.Client;
+
+namespace OpenTelemetry.Instrumentation.ServiceFabricRemoting;
+
+/// <summary>
+/// An IServiceRemotingClientFactory that uses Fabric TCP transport to create IServiceRemotingClient
+/// that communicate with actors over interfaces that are remoted via Microsoft.ServiceFabric.Services.Remoting.V2.FabricTransport.Runtime.FabricTransportServiceRemotingListener.
+///
+/// Both the client and the listener are instrumented to propagate both the traceContext and Baggage objects from OpenTelemetry
+/// </summary>
+public class TraceContextEnrichedActorRemotingClientFactory : IServiceRemotingClientFactory
+{
+    private readonly FabricTransportActorRemotingClientFactory innerFactory;
+
+    /// <summary>
+    ///  Initializes a new instance of the <see cref="TraceContextEnrichedActorRemotingClientFactory"/> class.
+    /// </summary>
+    /// <param name="fabricTransportRemotingSettings">The settings for the fabric transport. If the settings are not provided or null, default settings with no security.</param>
+    /// <param name="callbackMessageHandler">The callback client that receives the callbacks from the service.</param>
+    public TraceContextEnrichedActorRemotingClientFactory(FabricTransportRemotingSettings fabricTransportRemotingSettings, IServiceRemotingCallbackMessageHandler? callbackMessageHandler)
```

Comment:
Is it possible to avoid taking a dependency on the FabricTransport by defining the TraceContextEnriched\*ClientFactory classes as decorators for the IServiceRemotingClientFactory interface instead of the concrete FabricTransport*ClientFactory classes?

Reply:
Good idea. I made the change and added TraceContextEnrichedServiceRemotingClientFactoryAdapter.

Reply:
I don't think this was addressed. The factory still hardcodes the assumption that a FabricTransport is used for remoting.

Reply:
@olegsych , I think you are not looking at the latest version of the code. The new factory takes an IServiceRemotingClientFactory as a parameter in the constructor and doesn't take a dependency on FabricTransport.